pkg/agent/openflow/pipeline_windows.go

```diff
@@ -190,11 +200,78 @@ func (c *client) snatImplementationFlows(nodeIP net.IP, category cookie.Category
 	return flows
 }
 
+// serviceSNATFlows installs flows to allow a Windows host to access a ClusterIP Service.
+func (c *client) serviceSNATFlows(nodeIP net.IP, localGatewayMAC net.HardwareAddr, category cookie.Category) []binding.Flow {
+	var flows []binding.Flow
+	if c.enableProxy {
+		flows = append(flows, []binding.Flow{
+			// If the host uses another IP to access the Service, hostnetwork ClusterIP Service reply packet
+			// needs unSNAT with snatCTMark mark.
+			c.pipeline[uplinkTable].BuildFlow(priorityNormal).
+				MatchProtocol(binding.ProtocolIP).
+				MatchRegRange(int(marksReg), markTrafficFromUplink, trafficFromRegRange).
+				Action().CT(false, unSNATTable, ctZoneSNAT).NAT().CTDone().
```

Comment:
I feel back we do unNAT in uplinkTable (in two funcs), conntrackTable; and jump to different tables - unSNATTable, conntrackTable, conntrackStateTable; and output to bridge port in different tables after that; and worse in multiple funcs. How could others understand all the cases around NAT and uplink-bridge redirection?

Could we think about any way to unify the unNAT, SNAT, bridge redirection logics? @wenyingd : could you help here?

Reply:
I want to clarify one thing first: since we use two different ct_zone to perform DNAT and SNAT, it is difficult to perform all de-NAT in the same table. `unSNATTable` is for de-SNAT, while conntrackTable is for de-DNAT. Let's think about this scenario: a local Pod expects to access a Cluster IP Service, and the backend endpoint is a remote Pod which is using hostNetwork. The request packet on source Node should take DNAT in ct_zone=65520, and then take SNAT in ct_zone=65500. For the reply packet, it should take de-SNAT in ct_zone=65500 first, and then de-DNAT in ct_zone=65520. If we use a single table, it should be easy to introduce loop. 
I agree with you that current pipeline for the SNAT and external traffic is difficult to understand. I would try to re-orgnize the related logics.